### ISSUE TITLE
Multiple routes with the same name should not be able to exist

### ISSUE THREAD

**Issue by frankkoornstra** [chimeraphp/routing#32]
When adding multiple routes with the same name (but different paths) there's no error at "compile time", resulting in unexpected behavior on runtime. Just dropping this here for now for awareness :slightly_smiling_face:

**Comment by lcobucci** [comment]
Interesting, though I think this is related to underlying libraries and not chimera itself.

We can totally validate things on compile-time to not depend on specific implementations. I'll move this to the DI component and rephrase this as a new feature.

@frankkoornstra thanks :+1:
